crates/shared/src/order_validation.rs

```diff
@@ -480,8 +486,9 @@ impl OrderValidating for OrderValidator {
         }
 
         self.validity_configuration.validate_period(&order)?;
-
-        if has_same_buy_and_sell_token(&order, self.native_token.address()) {
+        if let Some(native_token) = &self.native_token
```

Comment:
Related to the [comment](https://github.com/cowprotocol/services/pull/3894#discussion_r2584494575) above, this would become `if let Some(native_token) = &self.enable_buy_equals_sell && has_same_buy_and_sell_token(&order, native_token.address())` which I think is kinda nice.

Reply:
You are right, after the changes it is even simpler as a single boolean flag controls it.